multiqc/modules/freyja/freyja.py

```diff
@@ -0,0 +1,141 @@
+        self.freyja_data = dict()
+
+        # Parse the output files
+        self.parse_summ_files()
+
+        # Remove filtered samples
+        self.freyja_data = self.ignore_samples(self.freyja_data)
+
+        # Let MultiQC know this module found no data
+        if len(self.freyja_data) == 0:
+            raise UserWarning
+
+        log.info(f"Found {len(self.freyja_data)} reports")
+        self.write_data_file(self.freyja_data, "multiqc_freyja")
+
+        self.general_stats_cols()
+        self.add_freyja_section()
+
+    def parse_summ_files(self):
+        """
+        Parse the summary file.
+        Freyja has multiple summary files, but we only need to parse the one from the demix command.
+        More specifically, we only need the line that starts with "summarized".
+        ...
+        summarized	[('BQ.1*', 0.983), ('Omicron', 0.011), ('key', value)]
+        ...
+        """
+        for f in self.find_log_files("freyja", filehandles=True):
+            s_name = f["s_name"]
+            # Read the statistics from file
+            d = {}
+            for line in f["f"]:
+                try:
+                    if line.startswith("summarized"):
+                        summarized_line = line
+                        summarized_line = summarized_line.strip().split("\t")[1]
+                        d = eval(
+                            summarized_line
+                        )  # Make sure no input is corrupted and does not contain any malicious code
+                        d = dict(d)
+                except ValueError:
+                    pass
+
+            # Percentages don't always add up to 1, show a warning if this is the case
+            if sum(d.values()) != 1:
+                log.warning(f"Freyja {s_name}: percentages don't sum to 1")
+
+            # There is no sample name in the log, so we use the root of the
+            # file as sample name (since the filename is always stats.dat
+            if s_name in self.freyja_data:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+            self.freyja_data[s_name] = d
+            self.add_data_source(f, s_name)
+
+    def general_stats_cols(self):
+        """Add a single column displaying the most abundant lineage to the General Statistics table"""
+        top_lineage_dict = {}
+        for s_name, sub_dict in self.freyja_data.items():
+            top_lineage = max(sub_dict, key=sub_dict.get)
+            top_lineage_value = sub_dict[top_lineage]
+            top_lineage_dict[s_name] = {
+                "Top_lineage_freyja": top_lineage,
+                "Top_lineage_freyja_percentage": top_lineage_value,
+            }
+
+        headers = OrderedDict()
+        headers["Top_lineage_freyja"] = {
+            "title": "Top lineage (Freyja)",
+            "description": "The most abundant lineage in the sample",
+            "scale": "RdYlGn-rev",  # Not sure if this is the best scale
```

Comment:
I think one of the categorical scales should be good here (like `Set2`, etc, see: https://multiqc.info/docs/development/plots/#table-colour-scales), though I can't make it work 🤔

Reply:
Ah, it's not working because the color is applied to the bar, which has a width of zero here.

Reply:
Okay, I pushed a change to use categorical colours instead, let me know if it looks good: 

<img width="996" alt="Screenshot 2023-08-30 at 15 26 43" src="https://github.com/ewels/MultiQC/assets/1575412/ce41c02c-6186-4f23-8a39-50338763109c">

<img width="999" alt="Screenshot 2023-08-30 at 15 25 48" src="https://github.com/ewels/MultiQC/assets/1575412/25826934-da19-4110-b1ba-625a4bcf28c6">
looks good

<img width="1005" alt="Screenshot 2023-08-30 at 15 24 50" src="https://github.com/ewels/MultiQC/assets/1575412/e8e6ba1d-d23f-4eaf-9660-7ecd596fa068">

And would love @ewels opinion on this.

Reply:
LGTM!